docs/arcadia-users-group/20230328-intro-to-python-3/lesson.md

```diff
@@ -0,0 +1,1123 @@
+
+---
+### 3.1 Plotting with **`matplotlib`**
+
+Presenting data in a visual format often helps us make decisions or communicate our results.  
+A popular Python package for making plots is **`matplotlib`**.
+
+Below, we use the module **`matplotlib.pyplot`** to plot some data using `x` and `y` coordinates.
+
+
+```python
+import matplotlib.pyplot as plt
+
+x = [1, 2, 3.5, 4.4, 6]
+y = [5, 4, 4.1, 3, 3.8]
+
+plt.plot(x, y)
+plt.show()
+```
+
+
+    
+![png](lesson_files/lesson_52_0.png)
+    
+
+
+You'll notice that rather than simply using `import matplotlib.pyplot`, we added an `as plt` expression to the end of our import statement.  
+Python allows you to import packages using aliases.  
+This saves you from having to type out `matplotlib.pyplot.plot()` and shortens the expression to `plt.plot()`.  
+
+Package documentation examples often gives you a suggestion for how you might abbreviate packages for others to be able to understand.  
+The base `matplotlib` package, for example, is often abbreviated as `mpl`.  
+You'll see additional examples of this in the other examples we'll look at.
+
+---
+### 3.2 Modifying a **`matplotlib`** plot
+
+So, we've plotted a basic line plot. But what do the axes mean? And why is it so large - can we make it smaller?  
+**`matplotlib`** is a popular package for plotting because you can customize essentially every single aspect of any plot you make.  
+
+See the comments below for an example of a more customized plot.
+
+
+```python
+import matplotlib.pyplot as plt
+
+x = [1, 2, 3.5, 4.4, 6]
+y = [5, 4, 4.1, 3, 3.8]
+
+# You can establish a figure and set the figure size prior to plotting.
+plt.figure(figsize = (3, 2)) 
+
+# You can change the background color of your plot
+ax = plt.axes()
+ax.set_facecolor("xkcd:midnight") # matplotlib recognizes a variety of color names
+
+plt.plot(x, y, c = 'c', linestyle = '-') # You can change the color (c) to cyan using 'c'. You can also set a dashed ('-') linestyle.
+
+# If you call another plotting function without creating a new figure, you can plot again on the same plot.
+plt.scatter(x, y, c = 'w', marker = '*', s = 50) # For a scatter plot, you can choose color (c), marker style, and marker size (s).
+
+# You can set the specific x and y limits of your plot
+plt.xlim(0.5, 6.5)
+plt.ylim(2.5, 5.5)
+
+# You can have matplotlib show a grid, and choose its color
+plt.grid(c = 'b')
+
+# You can label the x and y axes of the plot
+plt.xlabel('longitude')
+plt.ylabel('latitude')
+
+# You can add a title to the plot
+plt.title('Casseiopeia')
+
+plt.show()
+```
+
+
+    
+![png](lesson_files/lesson_54_0.png)
+    
+
+
+This example only scratches the barest surface of what you can do with **`plt`**.  
+Future workshops will dive into deeper detail about how **`matplotlib`** works and what you can do with it.
```

Comment:
```suggestion
If people want to learn more, we can have future workshops that dive into deeper detail about how **`matplotlib`** works and what you can do with it.
```

Reply:
you can keep it phrased as is, but I don't know enough about matplotlib to teach more about it and I didn't want you to feel locked in to covering it!